mgmtfn/mesosplugin/cniserver.go

```diff
@@ -0,0 +1,220 @@
+limitations under the License.
+*/
+
+package mesosplugin
+
+import (
+	"encoding/json"
+	"fmt"
+	log "github.com/Sirupsen/logrus"
+	"github.com/contiv/netplugin/mgmtfn/mesosplugin/cniapi"
+	"github.com/contiv/netplugin/netplugin/plugin"
+	"github.com/gorilla/mux"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+)
+
+type cniServer struct {
+	networkID      string
+	endpointID     string
+	endPointLabels map[string]string
+	ipv4Addr       string
+	pluginArgs     cniapi.CniCmdReqAttr
+	cniSuccessResp cniapi.CniCmdSuccessResp
+}
+
+type httpAPIFunc func(httpBody []byte) ([]byte, error)
+
+var cniLog *log.Entry
+
+// HTTP wrapper
+func httpWrapper(handlerFunc httpAPIFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+
+		if httpBody, err := ioutil.ReadAll(r.Body); err == nil {
+			cniLog.Infof("==== received new http req %s ======", string(httpBody))
+			jsonResp, hErr := handlerFunc(httpBody)
+			if hErr != nil {
+				cniLog.Infof("sending http Error response %s", string(jsonResp))
+				w.WriteHeader(http.StatusInternalServerError)
+			} else {
+				cniLog.Infof("sending http OK response %s", string(jsonResp))
+				w.WriteHeader(http.StatusOK)
+			}
+			w.Write(jsonResp)
+
+		} else {
+			errResp, _ := createCniErrorResponse(fmt.Errorf("failed to read http body: %s", err))
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write(errResp)
+		}
+		w.Header().Set("Content-Type", "application/json")
+	}
+}
+
+func createCniErrorResponse(errmsg error) ([]byte, error) {
+	// CNI_ERROR_UNSUPPORTED is sent for all errors
+	cniResp := cniapi.CniCmdErrorResp{CniVersion: cniapi.CNI_DEFAULT_VERSION,
+		ErrCode: cniapi.CNI_ERROR_UNSUPPORTED_FIELD,
+		ErrMsg:  fmt.Sprintf("Contiv: %s", errmsg.Error())}
+
+	cniLog.Infof("sending error response: %s", cniResp.ErrMsg)
+	jsonResp, err := json.Marshal(&cniResp)
+	if err != nil {
+		cniLog.Errorf("failed to convert error response to JSON: %s ", err.Error())
+		return nil, err
+	}
+	return jsonResp, errmsg
+
+}
+
+func (cniReq *cniServer) createCniSuccessResponse() ([]byte, error) {
+	jsonResp, err := json.Marshal(&cniReq.cniSuccessResp)
+	if err != nil {
+		cniLog.Errorf("failed to convert success response to JSON: %s ", err.Error())
+		return nil, err
+	}
+	return jsonResp, nil
+}
+
+// parse & save labels
+func (cniReq *cniServer) parseCniArgs(httpBody []byte) error {
+
+	if err := json.Unmarshal(httpBody, &cniReq.pluginArgs); err != nil {
+		return fmt.Errorf("failed to parse JSON req: %s", err.Error())
+	}
+
+	cniLog.Infof("parsed ifname name: %s", cniReq.pluginArgs.CniIfname)
+	cniLog.Infof("parsed netns name: %s", cniReq.pluginArgs.CniNetns)
+	cniLog.Infof("parsed container-id: %s", cniReq.pluginArgs.CniContainerid)
+	cniLog.Infof("parsed tenant name: %s", cniReq.pluginArgs.Labels.TenantName)
+	cniLog.Infof("parsed network name: %s", cniReq.pluginArgs.Labels.NetworkName)
+	cniLog.Infof("parsed network group: %s", cniReq.pluginArgs.Labels.NetworkGroup)
```

Comment:
Can we make these Debugf and log a single event for all the parameters received from mesos?